### PR TITLE
[Front-End] - Register Form

### PR DESCRIPTION
**Pull request by SGVGen**
## Ticket/s

- Register Page

## Changes

- Added Register Component
- Added Register Page
- Added Styling to Register Component
- Added Register Page Route to Rootview
- Added Header and Footer Navigation to RootView
- Modified Rootview CSS
- Modified StyledComponent CSS

## Screenshot

- image

## How to test?

- run **pnpm dev**
- After compilation is complete
- Insert to url **http://localhost:5173/register**

### CODE REVIEW

File: apps/web/src/components/auth/RegisterForm.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  PasswordInput,
+  Stack,
+  Text,
+  TextInput,
+  Input,
+  Group,
+  Flex,
+  Title,
+  Select,
+} from '@mantine/core';
```
Comment (DanDalanDone69) [inline apps/web/src/components/auth/RegisterForm.tsx]: remove unused import

File: apps/web/src/components/auth/RegisterForm.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+import * as z from 'zod';
+
+import { StyledContainer } from './styles';
+
+const schema = z.object({
+  email: z.string().min(1, 'Required'),
+  password: z.string().min(8, 'Minimum Password is 8'),
+});
+
+type RegisterValues = {
```
Comment (DanDalanDone69) [inline apps/web/src/components/auth/RegisterForm.tsx]:
```suggestion
type TRegisterValues = {
```

File: apps/web/src/components/auth/RegisterForm.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  email: string;
+  street: string;
+  region: string;
+  zipcode: number;
+  country: string;
+  password: string;
+  confirmpassword: string;
+};
+
+type RegisterFormProps = {
```
Comment (DanDalanDone69) [inline apps/web/src/components/auth/RegisterForm.tsx]:
```suggestion
type TRegisterFormProps = {
```

File: apps/web/src/components/auth/RegisterForm.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  onSuccess: () => void;
+};
+
+export const RegisterForm = ({ onSuccess }: RegisterFormProps) => {
```
Comment (DanDalanDone69) [inline apps/web/src/components/auth/RegisterForm.tsx]:
```suggestion
export const RegisterForm = ({ onSuccess }: TRegisterFormProps) => {
```

File: apps/web/src/components/auth/RegisterForm.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+                <Title order={1}>Register</Title>
+                <div style={{ display: 'none' }}>
+                  <Select
+                    value={data.values[0]}
+                    label="Creatable Select"
+                    data={data}
+                    placeholder="Select items"
+                    nothingFound="Nothing found"
+                    searchable
+                  />
```
Comment (DanDalanDone69) [inline apps/web/src/components/auth/RegisterForm.tsx]: What's the use of this if the display is none?

File: apps/web/src/styles/index.css [hunk 6]
```diff
@@ -1,7 +1,7 @@
 :root {
     max-width: 1280px;
```
Comment (DanDalanDone69) [inline apps/web/src/styles/index.css]: remove this

**Review by DanDalanDone69 — CHANGES_REQUESTED** [review]
Please, check some comments

File: apps/web/src/components/auth/RegisterForm.tsx (hunk 2 above)
Comment (SGVGen) [reply apps/web/src/components/auth/RegisterForm.tsx]: Added and Pushed to commit 93ef13d6c7a0956d3c3d1310b60d85fafa8b09c5.

File: apps/web/src/styles/index.css (hunk 6 above)
Comment (SGVGen) [reply apps/web/src/styles/index.css]: Added and Pushed to commit 93ef13d6c7a0956d3c3d1310b60d85fafa8b09c5.

File: apps/web/src/components/auth/RegisterForm.tsx (hunk 5 above)
Comment (SGVGen) [reply apps/web/src/components/auth/RegisterForm.tsx]: Added and Pushed to commit 93ef13d6c7a0956d3c3d1310b60d85fafa8b09c5.

File: apps/web/src/components/auth/RegisterForm.tsx (hunk 4 above)
Comment (SGVGen) [reply apps/web/src/components/auth/RegisterForm.tsx]: Added and Pushed to commit 93ef13d6c7a0956d3c3d1310b60d85fafa8b09c5.

File: apps/web/src/components/auth/RegisterForm.tsx (hunk 3 above)
Comment (SGVGen) [reply apps/web/src/components/auth/RegisterForm.tsx]: Added and Pushed to commit 93ef13d6c7a0956d3c3d1310b60d85fafa8b09c5.

**Comment by sandrex-SGV** [comment]
image

hello sir @SGVGen , please check attached img

**Comment by sandrex-SGV** [comment]
> image
> 
> hello sir @SGVGen , please check attached img

Resolved. Thanks

**Review by dandalandone — APPROVED** [review]
LGTM!

**Review by DanDalanDone69 — APPROVED** [review]

**Pull request merged**